cpp/src/parquet/file_writer.cc

```diff
@@ -290,14 +294,29 @@ class RowGroupSerializer : public RowGroupWriter::Contents {
       auto oi_builder = page_index_builder_ && properties_->page_index_enabled(path)
                             ? page_index_builder_->GetOffsetIndexBuilder(column_ordinal)
                             : nullptr;
+      auto bloom_filter =
+          bloom_filter_writer_ && properties_->bloom_filter_enabled(path)
+              ? bloom_filter_writer_->GetOrCreateBloomFilter(
+                    path,
+                    [&]() {
+                      int64_t ndv = properties_->bloom_filter_ndv(path);
+
+                      auto bloom_filter = std::make_unique<BlockSplitBloomFilter>();
+                      // TODO(mwish): add configurable ndv
+                      bloom_filter->Init(BlockSplitBloomFilter::OptimalNumOfBytes(
+                          static_cast<uint32_t>(ndv),
```

Comment:
So here we imply the ndv cannot exceed UINT32_MAX? Then we probably need to change its type in the writer properties.